### ISSUE TITLE
Disable the free kakera snipe

### ISSUE THREAD

**Issue by BellaDona007** [vivinano/MudaeAutoBot#73]
The bot is running the rolls perfectly, with few pauses, but there's still no way to disable the free kakera snipe.
To be perfect would need the "soulemoji_list" working.

Would it be possible for the bot to use $tu after claiming a waifu, so that it gets the new claim time?

**Comment by vivinano** [comment]
You can disable it via the code yourself its pretty simple to see the code that does the free kakera check 

in regards to soulemoji a new way has to be suggested to determine "soulmates" and the previous way was pretty inefficient. possible that im gonna remove the soulmate_emoji variable tho as it only used by the end game users and Users havent brought it up 

the bot already get the new claim timer based on the $settings 
![image](https://user-images.githubusercontent.com/33008397/125884346-a034e99b-9d92-4ad4-8d7a-fafd275e618a.png)

basically $tu has no difference in terms of whose $tu so if let say I type $tu the bot will now determine that my $tu is now ur new timer and will cause "False" positives hence making you never able to snipe because as long as one person rolls $tu then the bot will be set 

of course i could probably make it to where it could determine who rolled $tu but then again that is more computational power and you already had issues with "Slow" roll times 

I repeat though the current implementation is that if the bot finds a $settings it will get another claim timer

**Comment by BellaDona007** [comment]
> You can disable it via the code yourself its pretty simple to see the code that does the free kakera check
>

![image](https://user-images.githubusercontent.com/82736191/126004825-9e934cd0-2b5e-480c-91c6-55be377b09a7.png)

The only way I saw it would be, for an absurd amount here, but it would still click in time
 
> in regards to soulemoji a new way has to be suggested to determine "soulmates" and the previous way was pretty inefficient. possible that im gonna remove the soulmate_emoji variable tho as it only used by the end game users and Users havent brought it up
> 


Soulmate is not something endgame as you say, as it is possible for those starting today, to have 2 waifu with lvl 10 key.
Running the bot on a new server myself, I managed in just over 1 month 4 waifu key level 10+ starting from level 0.


> the bot already get the new claim timer based on the $settings
> ![image](https://user-images.githubusercontent.com/33008397/125884346-a034e99b-9d92-4ad4-8d7a-fafd275e618a.png)
> 
> basically $tu has no difference in terms of whose $tu so if let say I type $tu the bot will now determine that my $tu is now ur new timer and will cause "False" positives hence making you never able to snipe because as long as one person rolls $tu then the bot will be set
> 
What I've already noticed is that, if he uses a claim at the last minute, it will reset and won't use it in the next 2 hours, I'm forced to close the bot every time it takes a waifu, so that it takes the reset time

**Comment by vivinano** [comment]
The changes to the code you made doesnt change anything for the 1st one i recommend if you dont enjoy it just remove the whole function

I do not have a soulmate and I have no ability to personally test any soulmate sniping implementation so If your "Developer" can not test a function how does the Developer make a system to share. I was literally shooting in the dark with soul mate sniping.

the claim window is set automaticlly based on the $settings if you $settings are invalid or are not being read from a file then it will not be appropriate set the new claim window

**Comment by BellaDona007** [comment]
> The changes to the code you made doesnt change anything for the 1st one i recommend if you dont enjoy it just remove the whole function
> 

I thought if I turned it off it would stop working perfectly, thanks for the reply

> I do not have a soulmate and I have no ability to personally test any soulmate sniping implementation so If your "Developer" can not test a function how does the Developer make a system to share. I was literally shooting in the dark with soul mate sniping.
> 

It is not necessary to have soulmater to test, just have key lvl 1, because kakera's claim would be based on embedcolor

> the claim window is set automaticlly based on the $settings if you $settings are invalid or are not being read from a file then it will not be appropriate set the new claim window


It happened now, claim missing 1h to reset.
A character appeared after the reset and he didn't catch. I had to close and reopen the bot.
I will use $settings again and delete the existing file

**Comment by vivinano** [comment]
![image](https://user-images.githubusercontent.com/33008397/126241167-3d877793-ca73-433c-8d83-6652f7a42fe7.png)

the last minute claim is working fine if you dont want it either edit your `Last_claim_window` or turn true to false as testing it for 1 day it has been working fine. The last minute claim window was added to use a claim there is a risk to it so there is a understanding that it might claim something that is not what you want for the sake of not wasting a claim.

show me when the bot claimed, a list of the hours that resets happen ex: 
3:56
6:56
9:56
12:56

and the claim that it ignored , $settings and channeldata pertaining to that channel

you can implement it back in based on a previous version but manually having to set embedcolor for each key'd character means that its no longer just start and forget about it so until a better way is known I will not be pushing a commit to add this feature back as this will just add more "checks" which will slow things down

**Comment by BellaDona007** [comment]
> ![image](https://user-images.githubusercontent.com/33008397/126241167-3d877793-ca73-433c-8d83-6652f7a42fe7.png)
> 
> the last minute claim is working fine if you dont want it either edit your `Last_claim_window` or turn true to false as testing it for 1 day it has been working fine. The last minute claim window was added to use a claim there is a risk to it so there is a understanding that it might claim something that is not what you want for the sake of not wasting a claim.
> 
> show me when the bot claimed, a list of the hours that resets happen ex:
> 3:56
> 6:56
> 9:56
> 12:56
> 
> and the claim that it ignored , $settings and channeldata pertaining to that channel
> 

I will test for 24h


> you can implement it back in based on a previous version but manually having to set embedcolor for each key'd character means that its no longer just start and forget about it so until a better way is known I will not be pushing a commit to add this feature back as this will just add more "checks" which will slow things down

![image](https://user-images.githubusercontent.com/82736191/126253842-858c0d0d-0379-400e-8066-c6c17659c071.png)


Unfortunately the only quick way to identify your 10+ key would be to use these items:

**1** (chaoskey) linked to, **2** (Owner) 
**3** embedcolor

**Comment by vivinano** [comment]
in terms of kakera theoretically it could be added here:

![image](https://user-images.githubusercontent.com/33008397/126739773-98e9b00a-19e7-4fe9-b952-b2f4c2bfbcf7.png)

but it would have to run react_m which is a get_message call on every roll twice which will slow thing down even more. and you already had issues with rolling taking to long. I can't see a way to add without double checking 1 message for both conditions

**Comment by vivinano** [comment]
alright so the new commit should add it back but no in the setting json. you will have to make all changes in code and basically uncomment out all the code because I dont want to implement it without testing if it makes anything else slower

![image](https://user-images.githubusercontent.com/33008397/126741570-461aad15-4be1-4918-b9ad-a6d3208dcb32.png)
![image](https://user-images.githubusercontent.com/33008397/126741580-6f7f5dca-99bb-4338-b535-d00485f2a3fd.png)

basically remove the `#` on every line of the changes

**Comment by BellaDona007** [comment]
thank you very much, i will test
